job-server/src/spark.jobserver/JobInfoActor.scala

```diff
@@ -33,38 +33,39 @@ class JobInfoActor(jobDao: JobDAO, contextSupervisor: ActorRef) extends Instrume
 
   override def wrappedReceive: Receive = {
     case GetJobStatuses(limit) =>
-      sender ! jobDao.getJobInfos(limit.get)
+      val originator = sender
+      jobDao.getJobInfos(limit.get).foreach(originator ! _)
 
     case GetJobStatus(jobId) =>
-      val jobInfo = Await.result(jobDao.getJobInfo(jobId), 60 seconds)
-      val resp = if (jobInfo.nonEmpty) NoSuchJobId else jobInfo.get
-      sender ! resp
+      val originator = sender
+
+      jobDao.getJobInfo(jobId).collect {
+        case Some(jobInfo) => originator ! jobInfo
+        case None => originator ! NoSuchJobId
+      }
 
     case GetJobResult(jobId) =>
-      breakable {
-        val jobInfo: Option[JobInfo] = Await.result(jobDao.getJobInfo(jobId), 60 seconds)
-
-        if (jobInfo.nonEmpty) {
-          sender ! NoSuchJobId
-          break
-        }
-
-        jobInfo.filter { job => job.isRunning || job.isErroredOut }
-          .foreach { jobInfo =>
-            sender ! jobInfo
-            break
-          }
+      val originator = sender
 
-        // get the context from jobInfo
-        val context = jobInfo.get.contextName
+      jobDao.getJobInfo(jobId).collect {
+        case Some(jobInfo) =>
+          if (jobInfo.isRunning || jobInfo.isErroredOut)
+          {
```

Comment:
(style) let's move the braces to be with the other lines, like

    if (jobInfo.isRunning || jobInfo.isErroredOut) {
      originator ! jobInfo
    } else {